src/SimulationRunner.hh

```diff
@@ -160,9 +180,29 @@ namespace ignition
 
       /// \brief Add system after the simulation runner has been instantiated
       /// \note This actually adds system to a queue. The system is added to the
-      /// runner at the begining of the a simulation cycle (call to Run)
+      /// runner at the begining of the a simulation cycle (call to Run). It is
+      /// also responsible for calling `Configure` on the system.
+      /// \param[in] _system SystemPluginPtr to be added
+      /// \param[in] _entity Entity that system is attached to. If nullopt,
+      /// system is attached to a world.
+      /// \param[in] _sdf Pointer to the SDF of the entity. Nullopt defaults to
+      /// SDF of the entire world.
+      public: void AddSystem(const SystemPluginPtr &_system,
+          std::optional<Entity> _entity = std::nullopt,
+          std::optional<sdf::ElementPtr> _sdf = std::nullopt);
+
+      /// \brief Add system after the simulation runner has been instantiated
+      /// \note This actually adds system to a queue. The system is added to the
+      /// runner at the begining of the a simulation cycle (call to Run). It is
+      /// also responsible for calling `Configure` on the system.
       /// \param[in] _system System to be added
-      public: void AddSystem(const SystemPluginPtr &_system);
+      /// \param[in] _entity Entity of system to be added. Nullopt if system
+      /// doesn't connect to an entity.
+      /// \param[in] _sdf Pointer to the SDF of the entity. Nullopt defaults to
+      /// world.
+      public: void AddSystem(System *_system,
+          std::optional<Entity> _entity = std::nullopt,
+          std::optional<sdf::ElementPtr> _sdf = std::nullopt);
```

Comment:
What do you think about using `std::shared_ptr<const Element>` (a.k.a `sdf::ElementConstPtr` after the next libsdformat release) here so that code that only has access to `const Element` would be able to call this. For example, if a system wants to add another system during `System::Configure`, it wouldn't be able to call this without a `const_cast`.

Reply:
> using `std::shared_ptr<const Element>`

Done in d8c6047

> if a system wants to add another system during System::Configure

Note that this isn't possible right now, since systems don't have access to the `SimulationRunner` or `Server` APIs. Unless they spawn a new server 😳😵

Reply:
> Note that this isn't possible right now, since systems don't have access to the SimulationRunner or Server APIs. Unless they spawn a new server

Not sure why I thought it was possible. I guess a contrived scenario is where a test creates its own `Server` and passes it to an internal `System`.